include/revng-c/mlir/Dialect/Clift/IR/CliftOps.td

```diff
@@ -26,6 +27,120 @@ def Clift_StatementRegion : Region<CPred<"::mlir::clift::impl::verifyRegionConte
 def Clift_ExpressionRegion : Region<CPred<"::mlir::clift::impl::verifyRegionContent<::mlir::clift::ExpressionOpInterface>($_self)">,
                                     "Region representing an expression">;
 
+//===------------------------ Top level operations ------------------------===//
+
+def Clift_ModuleOp : Clift_Op<"module",
+                              [SymbolTable,
+                               HasOnlyGraphRegion,
+                               NoRegionArguments,
+                               NoTerminator,
+                               SingleBlock,
+                               RegionKindInterface]> {
+  let regions = (region AnyRegion:$body);
+
+  let skipDefaultBuilders = 1;
```

Comment:
I know this was present even before your changes, but can you please refresh what it does and why we need it?

Reply:
It disables generation of the default builders for this operation. The builders are functions used by `mlir::OpBuilder` when you do e.g. `builder.create<ModuleOp>(loc)`. We want to define our own builder without arguments to give it a custom body. You can see that [here](https://github.com/vasama-work/revng-c/blob/feature/clift-ops/lib/mlir/Dialect/Clift/IR/CliftOps.cpp#L53). The purpose of it is to ensure that a module created in such a way always has a block in its sole region.

Reply:
Cool thanks!